### PR TITLE
React: Add why-did-you-render dev extension to better see what components re-render and why (in console)

### PR DESCRIPTION
**Pull request by torkelo**
Uses: https://www.npmjs.com/package/@welldone-software/why-did-you-render  

Only enabled in dev mode and if you start grafana with "&why-render" url query parameter. 

This will automatically log in console what pure components (and FC that use memo) render and why. But for other functional components, you need to add this:

MyComponent.whyDidYouRender = true

You can also enable additional logging:

MyComponent.whyDidYouRender = {
  logOnDifferentValues: true,
  customName: 'Menu'
}

Example:

image

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
LGTM, great addition!

**Comment by mckn** [comment]
This is awesome!

**Review by dprokop — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -211,6 +211,7 @@
     "@types/md5": "^2.1.33",
     "@types/react-loadable": "5.5.2",
     "@types/react-virtualized-auto-sizer": "1.0.0",
+    "@welldone-software/why-did-you-render": "^4.0.6",
```
Comment (dprokop) [inline package.json:214]: should we fix version?

**Comment by torkelo** [comment]
For add buttons we should not use plus-circle, only plus, looks so much better

**Pull request merged**